### PR TITLE
Admin views for creating orginaztions and climate reports

### PR DESCRIPTION
**Pull request by emmabckstrm**
## NOTE!

This PR changes the order of the migrations, meaning that before running this migration, production db needs to rollback two steps.

### CODE REVIEW

File: app/controllers/admin/climate_reports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    end
+
+    def create
+      @report = ClimateReports::Report.new(report_params)
+
+      if @report.save
+        redirect_to [:admin, @report], notice: 'Report was successfully created.'
+      else
+        render :new
+      end
```
Comment (stefanlindbohm) [inline app/controllers/admin/climate_reports_controller.rb:29]: I would do this as a guard clause for the non-success path:

```ruby
unless @report.save
  render :new
  return
end

redirect_to [:admin, @report], notice: 'Report was successfully created.'
```

File: app/controllers/admin/climate_reports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    end
+
+    def update
+      if @report.update(report_params)
+        redirect_to [:admin, @report], notice: 'Report was successfully updated.'
+      else
+        render :edit
+      end
```
Comment (stefanlindbohm) [inline app/controllers/admin/climate_reports_controller.rb:37]: Same as above.

File: app/controllers/admin/climate_reports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    def set_organizations
+      @organizations = Organization.all.order(name: :asc).map { |o| [o.name, o.id] }
+    end
+
+    def report_params
+      params.require(:climate_report).permit(:title, :start_date, :end_date, :organization).tap do |p|
+        p[:reporting_period] = p[:start_date]..p[:end_date]
+        p[:organization_id] = p[:organization]
+      end.except(:start_date, :end_date, :organization)
```
Comment (stefanlindbohm) [inline app/controllers/admin/climate_reports_controller.rb:60]: Nice solution 😏

Comment (stefanlindbohm) [reply app/controllers/admin/climate_reports_controller.rb:60]: ...but why isn't `organization` just named `organization_id` in all places?

File: app/controllers/admin/organizations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    end
+
+    def create
+      @organization = Organization.new(organization_params)
+
+      if @organization.save
+        redirect_to [:admin, @organization], notice: 'Organization was successfully created.'
+      else
+        render :new
+      end
```
Comment (stefanlindbohm) [inline app/controllers/admin/organizations_controller.rb:28]: I'd prefer a guard clause here too

File: app/controllers/admin/organizations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    end
+
+    def update
+      if @organization.update(organization_params)
+        redirect_to [:admin, @organization], notice: 'Organization was successfully updated.'
+      else
+        render :edit
+      end
```
Comment (stefanlindbohm) [inline app/controllers/admin/organizations_controller.rb:36]: Once more.

**Comment by emmabckstrm** [comment]
@stefanlindbohm 's comments are addressed in https://github.com/goclimate/goclimate/commit/645be389ca6e9a7e662ce6f1ca776a0f5ba9fe89

**Pull request merged**